### PR TITLE
CupertinoTabBar: Add clickable cursor on web

### PR DESCRIPTION
**Pull request by TahaTesser**
partial fix for https://github.com/flutter/flutter/issues/86174

Complete details https://github.com/flutter/flutter/issues/86174#issuecomment-1016596648

### CupertinoTabBar
```console
Launching integration_test/cupertino/cupertino_tab_bar.dart on Chrome in debug mode...
integration_test/cupertino/cupertino_tab_bar.dart:1
DartUri: Unresolved uri: dart:web_sql
DartUri: Unresolved uri: dart:ui

This app is linked to the debug service: ws://127.0.0.1:55352/htWEd61mxMk=/ws
Debug service listening on ws://127.0.0.1:55352/htWEd61mxMk=/ws
💪 Running with sound null safety 💪
Connecting to VM Service at ws://127.0.0.1:55352/htWEd61mxMk=/ws
00:00 +0: Hovering over tab bar item updates cursor to clickable on Web
00:00 +1: (tearDownAll)
00:00 +2: All tests passed!
```
 
## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by Renzo-Olivares — COMMENTED** [review]
Similar to my review on [`CupertinoButton`](https://github.com/flutter/flutter/pull/96863#pullrequestreview-898169161). A test for when the cursor is not hovering on top of the button would be good.

File: packages/flutter/test/cupertino/bottom_tab_bar_test.dart [hunk 1]
```diff
@@ -568,4 +570,37 @@ Future<void> main() async {
         decoratedBoxHiddenBorder.decoration as BoxDecoration;
     expect(boxDecorationHiddenBorder.border, isNull);
   });
+
+  testWidgets('Hoveing over Bar item updates cursor to clickable', (WidgetTester tester) async {
```
Comment (Renzo-Olivares) [inline packages/flutter/test/cupertino/bottom_tab_bar_test.dart:574]: `Hoveing` -> `Hovering`, and `Bar` -> `tab bar`.

File: packages/flutter/test/cupertino/bottom_tab_bar_test.dart [hunk 2]
```diff
@@ -568,4 +570,37 @@ Future<void> main() async {
...
+            icon: ImageIcon(MemoryImage(Uint8List.fromList(kTransparentImage))),
+            label: 'Tab 1',
+          ),
+          BottomNavigationBarItem(
+            icon: ImageIcon(MemoryImage(Uint8List.fromList(kTransparentImage))),
+            label: 'Tab 2',
+          ),
+        ],
+      ),
+    ));
```
Comment (Renzo-Olivares) [inline packages/flutter/test/cupertino/bottom_tab_bar_test.dart:589]: Nit: trailing comma for formatting.

```dart
await pumpWidgetWithBoilerplate(
    tester,
    MediaQuery(
    ...
    ...
    ),
);
```

**Review by Renzo-Olivares — CHANGES_REQUESTED** [review]

**Comment by TahaTesser** [comment]
@Renzo-Olivares 
Thanks so much for the review, appreciate it! 

I intend to make similar changes to other Cupertino widgets ASAP.  I wanted to some confirmation with these two current  PRs.

**Comment by TahaTesser** [comment]
@Renzo-Olivares 
Actually tested the test on the web using integration_test, it works as expected, results in the description.

**Review by Renzo-Olivares — APPROVED** [review]
LGTM!

**Pull request merged**